os/mm/mm_heap/mm_partition_mgr.c

```diff
@@ -115,24 +117,27 @@ void mm_initialize_ram_partitions(void)
 
 int8_t mm_allocate_ram_partition(uint32_t **start_addr, uint32_t *size)
 {
-	int i;
-
-	for (i = 0; i < CONFIG_NUM_APPS; i++) {
-		if (g_mempartitions[i].status == MM_PART_FREE) {
-			g_mempartitions[i].status = MM_PART_USED;
-			*start_addr = (uint32_t *)g_mempartitions[i].start;
-			*size = g_mempartitions[i].size;
-			mvdbg("Allocated partition Start = 0x%x Size = %u bytes\n", *start_addr, *size);
+	if (*size == 0) {
+		*size = g_default_size;
+	}
 
-			/* struct mm_heap_s will be situated at start of partition
-			and heap will be initialized after this */
-			mm_initialize((struct mm_heap_s *)*start_addr, *start_addr + sizeof(struct mm_heap_s), *size - sizeof(struct mm_heap_s));
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+	ARCH_GET_RET_ADDRESS
+	*start_addr = mm_malloc(&g_pheap, *size + sizeof(struct mm_heap_s), retaddr);
+#else
+	*start_addr = mm_malloc(&g_pheap, *size + sizeof(struct mm_heap_s));
+#endif
 
-			return OK;
-		}
+	if (*start_addr == NULL) {
+		mdbg("Failed to allocate RAM partition of size %u\n", *size);
+		return -ENOMEM;
 	}
 
-	return ERROR;
+	/* struct mm_heap_s will be situated at start of partition and heap will be initialized after this */
+	mm_initialize((struct mm_heap_s *)*start_addr, (uint8_t *)*start_addr + sizeof(struct mm_heap_s), *size);
```

Comment:
Will the initialization be performed again here other than mm_initialize_ram_partitions ?

Reply:
Yes. We have two heaps here. 
1. Partition heap: g_pheap manages all the partitions
2. App partition heap: Each app partition will have its own heap